dlib/test/dnn.cpp

```diff
@@ -2486,6 +2486,32 @@ namespace
         }
     }
 
+    void test_serialization()
+    {
+        print_spinner();
+
+        using net_type = loss_mean_squared<fc<1, input<matrix<double>>>>;
+        net_type net, net2;
+
+#if 0
+        // At least on MSVC, this fails with compiler error:
+        // "C2665: 'dlib::serialize': none of the 46 overloads could convert all the argument types"
+        {
+            std::ostringstream out;
+            dlib::serialize(net, out);
```

Comment:
That code should compile. It's probably a bug in visual studio. There are often name lookup bugs in that compiler of this sort. The thing to do is to try and figure out why it's refusing to call serialize rather than add some proxy. 

You should check in gcc and see what happens. I would do it now but I only have my phone :/